src/toniarts/openkeeper/game/state/MainMenuScreenController.java

```diff
@@ -828,4 +830,76 @@ private void showBriefing() {
             }
         }
     }
+
+    public void showDebriefing(GameResult result) {
+        goToScreen(SCREEN_DEBRIEFING_ID);
+
+        Label levelTitle = screen.findNiftyControl("dLevelTitle", Label.class);
+
+        Element mainObjectiveImage = screen.findElementById("dMainObjectiveImage");
+        Element subObjectiveImage = screen.findElementById("dSubObjectiveImage");
+
+        String objectiveImage = String.format("Textures/Obj_Shots/%s-$index.png", state.selectedLevel.getFileName());
+        NiftyImage img = null;
+        GameLevel gameLevel = state.selectedLevel.getKwdFile().getGameLevel();
+        levelTitle.setText(gameLevel.getTitle());
+
+        try {
+            img = nifty.createImage(objectiveImage.replace("$index", "0"), false);
+            mainObjectiveImage.getRenderer(ImageRenderer.class).setImage(img);
+            mainObjectiveImage.setWidth(img.getWidth());
+            mainObjectiveImage.setHeight(img.getHeight());
+            mainObjectiveImage.show();
+        } catch (Exception e) {
+            logger.warning("Can't find image " + objectiveImage.replace("$index", "0"));
+            mainObjectiveImage.hide();
+        }
+
+        Label totalEvilRating = screen.findNiftyControl("totalEvilRating", Label.class);
+        Label overallTotalEvilRating = screen.findNiftyControl("overallTotalEvilRating", Label.class);
+        Label specialsFound = screen.findNiftyControl("specialsFound", Label.class);
+
+        subObjectiveImage.hide();
+        if (state.selectedLevel instanceof Level
+                && ((Level) state.selectedLevel).getType().equals(Level.LevelType.Level)) {
+            try {
+                img = nifty.createImage(objectiveImage.replace("$index", "1"), false);
+                subObjectiveImage.getRenderer(ImageRenderer.class).setImage(img);
+                subObjectiveImage.setWidth(img.getWidth());
+                subObjectiveImage.setHeight(img.getHeight());
+                subObjectiveImage.show();
+            } catch (Exception e) {
+                logger.warning("Can't find image " + objectiveImage.replace("$index", "1"));
+                subObjectiveImage.hide();
+            }
+        }
+
+        boolean levelWon = result.getData(GameResult.ResultType.LEVEL_WON);
+        screen.findNiftyControl("levelWon", Label.class).setText(levelWon ? "${menu.21}" : "${menu.22}");
+        int timeTaken = (int) result.getData(GameResult.ResultType.TIME_TAKEN);
+        screen.findNiftyControl("timeTaken", Label.class).setText(timeToString(timeTaken));
+    }
+
+    private String timeToString(int time) {
+        String result = "";
```

Comment:
I am not sure that it will give the same result as my function (outputs like in the original game). If DateTimeFormatter not output days if days = 0, hours - if 0, minutes - if 0, than I can use it.